### DISCUSSION TITLE
Streaming h264

### DISCUSSION THREAD

**Discussion by EricHenry** [webrtc-rs/webrtc#438] [Q&A]
I'm reading in a raw h264 stream from a camera, since the stream is already h264 encoded I am passing it along directly to the H264Reader and sending over a TrackLocalStaticSample track. However, the client on the opposite side is just rendering a black screen. Any tips in debugging if the stream is in the correct format?

**Comment by fantasyland101** [comment]
I am pretty mutch a beginer so take this with a grane of salt:
I would try to open the stream in vlc or somthing similar before it is opend by webrtc. If you open the stream as a h264 stream it should display your stream if it is h264.

when you create a webrtc stream they share a RTC session decription. this is a bunch of text and inside it their is information of in what format the video should be sent in.

her is how i view it:
```
socket.addEventListener("message", (event) => {
	pc.setRemoteDescription(new RTCSessionDescription(JSON.parse(atob(event.data))))
	console.log(JSON.parse(atob(event.data)));  //prints data into the console
});
```
In my case i have a VP8 somewhere in the text.

Finaly maybe you can use wireshark 
i go into my loopbac trafic capture and then i can se udp packets that are being sent (the video packets) 
and if they are recived by the browser :)

**Comment by shiqifeng2000** [comment]
> I'm reading in a raw h264 stream from a camera, since the stream is already h264 encoded I am passing it along directly to the H264Reader and sending over a TrackLocalStaticSample track. However, the client on the opposite side is just rendering a black screen. Any tips in debugging if the stream is in the correct format?



> I'm reading in a raw h264 stream from a camera, since the stream is already h264 encoded I am passing it along directly to the H264Reader and sending over a TrackLocalStaticSample track. However, the client on the opposite side is just rendering a black screen. Any tips in debugging if the stream is in the correct format?

if you send h264 es-stream using webrtc, make sure you send it in annex-b format, which means you will need 0001 as the startcode, including sps/pps before every IDR frame and so on for the latter nal packet
